Cargo.toml

```diff
@@ -16,16 +16,18 @@ repository = "https://github.com/zbraniecki/tinystr"
 readme = "README.md"
 keywords = ["string", "str", "small", "tiny", "no_std"]
 categories = ["data-structures"]
+# needed to avoid dev-dependency features unifying
+resolver = "2"
 
 [dependencies]
-serde = { version = "1.0.123", optional = true }
+serde = { version = "1.0.123", optional = true, default-features = false, features = ["alloc"] }
```

Comment:
Can you get the TinyStr Serde to be no-std?

Reply:
What do you mean?

Reply:
I mean no_std and no alloc.  Error handling aside, that should work, right?

Reply:
Oh. We need it for human-readable deserializing

Reply:
I think it would be great if we could feature-gate that somehow so that we can build TinyStr Serde with no alloc, even if it only supports Bincode and not JSON.

Reply:
Alternatively, is there a way to rewrite the human-readable Serde to be no-alloc?  My understanding is that it needs to be alloc because strings could have escape sequences.  Could we loop over the string character by character instead?

Reply:
Do we need no-alloc no-std, however? We're  going to need that anyway all throughout ICU4X

Reply:
So yes, we can have no-alloc tinystr which doesn't support non-human-readable deserialization. However I'm not sure how useful that is given that bincode strongly requires std by design (it depends on IO stuff directly)

Reply:
Actually maybe it doesn't, but anyway, I think that this makes more sense as something to be fixed in the future

Reply:
Follow-up is fine.  But what did you think about rewriting the human-readable Serde to loop over characters?  Is there a way to do that?

Reply:
Strings are a separate serde "type", backends are allowed to internally represent them however they want. From serde's POV it would be totally find for a backend to take the string and serialize it in reverse order or whatever as long as the deserializer does the right thing.